src/test/scala/org/broadinstitute/dsde/workbench/sam/service/ResourceServiceSpec.scala

```diff
@@ -1418,4 +1427,206 @@ class ResourceServiceSpec extends AnyFlatSpec with Matchers with ScalaFutures wi
       AccessPolicyDescendantPermissions(defaultResourceType.name, Set(ResourceAction("I don't exist either")), Set(ResourceRoleName("I don't exist"))),
     )).unsafeRunSync().size shouldBe 4
   }
+
+  "createAccessChangeEvents" should "show not changes for empty beforePolicies and afterPolicies" in {
+    val resource = genResource.sample.get
+    val beforePolicies = List.empty
+    val afterPolicies = List.empty
+    val result = service.createAccessChangeEvents(resource.fullyQualifiedId, beforePolicies, afterPolicies)
```

Comment:
This method doesn't seem to need any of the `service`'s state. Could you make it a free function so that one can call it without needing a `service` instance?

Reply:
it is an internal utility function of the service. Is there a benefit to setting it free?

Reply:
I'm of the camp it's always better to set free. More often than not in this codebase I've wanted to re-use private utility methods to the point where I'm not sure what making something private achieves other than documenting "this is not unit tested". I also prefer free functions over methods as they're easier to re-use.